File: DirectoryListing/app/src/main/java/com/example/directorylisting/auth/OkHttpUtils.java
/*
 * Copyright © 2017 Michael Steele.
 * All rights reserved.
 */

package com.example.directorylisting.auth;


import android.content.Context;

import com.example.directorylisting.shared.AppManager;

import java.security.cert.CertificateException;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLSession;
import javax.net.ssl.SSLSocketFactory;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;

import okhttp3.OkHttpClient;

public class OkHttpUtils {

    public static OkHttpClient.Builder getUnsafeOkHttpClientBuilder(final Context context, OkHttpClient client) {
        try {

            // Create a trust manager that does not validate certificate chains
            final TrustManager[] trustAllCerts = new TrustManager[] {
                    new X509TrustManager() {
                        @Override
                        public void checkClientTrusted(java.security.cert.X509Certificate[] chain, String authType) throws CertificateException {
                        }

                        @Override
                        public void checkServerTrusted(java.security.cert.X509Certificate[] chain, String authType) throws CertificateException {
                        }

                        @Override
                        public java.security.cert.X509Certificate[] getAcceptedIssuers() {
                            return new java.security.cert.X509Certificate[]{
                                    SSLUtils.getX509Certificate(context, AppManager.shared.webService.getCert(context))
                            };
                        }
                    }
            };

            SSLContext sslContext = SSLUtils.getSslContextForCertificateFile(context, trustAllCerts, SSLUtils.getKeyStore(context, SSLUtils.getX509Certificate(context, AppManager.shared.webService.getCert(context))));
            final SSLSocketFactory sslSocketFactory = sslContext.getSocketFactory();

            OkHttpClient.Builder builder = client.newBuilder();
            builder.sslSocketFactory(sslSocketFactory, (X509TrustManager)trustAllCerts[0]);
            builder.hostnameVerifier(new HostnameVerifier() {
                @Override
                public boolean verify(String hostname, SSLSession session) {
                    return true;
                }
            });

            return builder;
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: DirectoryListing/app/src/main/java/com/example/directorylisting/auth/OkHttpUtils.kt
/*
 * Copyright © 2017 Michael Steele.
 * All rights reserved.
 */

package com.example.directorylisting.auth


import android.content.Context
import com.example.directorylisting.api.WebService
import okhttp3.OkHttpClient
import java.security.cert.CertificateException
import javax.net.ssl.TrustManager
import javax.net.ssl.X509TrustManager

object OkHttpUtils {

    fun getUnsafeOkHttpClientBuilder(context: Context, client: OkHttpClient): OkHttpClient.Builder {
        try {

            // Create a trust manager that does not validate certificate chains
            val trustAllCerts = arrayOf<TrustManager>(object : X509TrustManager {
                @Throws(CertificateException::class)
                override fun checkClientTrusted(chain: Array<java.security.cert.X509Certificate>, authType: String) {
                }

                @Throws(CertificateException::class)
                override fun checkServerTrusted(chain: Array<java.security.cert.X509Certificate>, authType: String) {
                }

                override fun getAcceptedIssuers(): Array<java.security.cert.X509Certificate> {
                    return arrayOf(SSLUtils.getX509Certificate(context, WebService.getCert(context))) //AppManager.shared.webService
                }
            })

            val sslContext = SSLUtils.getSslContextForCertificateFile(context, trustAllCerts, SSLUtils.getKeyStore(context, SSLUtils.getX509Certificate(context, WebService.getCert(context)))) //AppManager.shared.w
            val sslSocketFactory = sslContext.socketFactory

            val builder = client.newBuilder()
            builder.sslSocketFactory(sslSocketFactory, trustAllCerts[0] as X509TrustManager)
            builder.hostnameVerifier { hostname, session -> true }

            return builder
        } catch (e: Exception) {
            throw RuntimeException(e)
        }

    }
}